tests/integration/operator/deploy_utils.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	"maistra.io/istio-operator/api/v1alpha1"
+	"maistra.io/istio-operator/pkg/util/tests/helm"
+	"maistra.io/istio-operator/pkg/util/tests/kubectl"
+	"sigs.k8s.io/yaml"
+)
+
+// deployOperator deploys the operator to either an OpenShift cluster or a Kubernetes cluster based on the value of the 'ocp' variable.
+// The operator will be deployed in the namespace specified by the 'namespace' variable.
+func deployOperator() error {
+	extraArg := ""
+	if ocp == "true" {
+		extraArg = "--set=platform=openshift"
+	}
+
+	output, err := helm.Template("chart", filepath.Join(baseDir, "chart"), namespace, "--include-crds", fmt.Sprintf("--set=image=%s", image), extraArg)
+	if err != nil {
+		return err
+	}
+
+	err = kubectl.ApplyString(output)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// undeployOperator is a function that undeploys the operator from either an OpenShift cluster or a Kubernetes cluster.
+// If the 'ocp' variable is set to "true", the operator will be undeployed from the OpenShift cluster.
+// Otherwise, it will be undeployed from the Kubernetes cluster.
+func undeployOperator() error {
+	extraArg := ""
+	if ocp == "true" {
+		extraArg = "--set=platform=openshift"
+	}
+
+	output, err := helm.Template("chart", fmt.Sprintf("%s/chart", baseDir), namespace, "--include-crds", fmt.Sprintf("--set=image=%s", image), extraArg)
+	if err != nil {
+		return err
+	}
+
+	err = kubectl.DeleteString(output)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createIstioCR create the istio CR for the specified version.
+// The control panel will be installed in the namespace specified by the 'controlPlaneNamespace' variable.
+// TODO: the controlPlaneNamespace variable is not been replaced in the source file, by default is set to istio-system.
+func createIstioCR(version string) error {
+	yamlString, err := readAndReplaceVersionInManifest(version)
+	if err != nil {
+		GinkgoWriter.Println("Error updating Istio manifest:", err)
+		return err
```

Comment:
Why not just wrap the error instead of printing to the log? In general, functions shouldn't log errors if they return them. This way, the caller can decide what to do with the error.

Reply:
Sorry, I added this for debugging while I was checking the errors applying some manifest